test/widgets/lightbox_test.dart

```diff
@@ -152,5 +267,148 @@ void main() {
+      checkPosition(tester, halfTime);
+      check(FakeVideoPlayerPlatform.position).equals(halfTime);
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+
+      // Near the end of the video.
+      await tester.pump(halfTime - const Duration(milliseconds: 500));
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester,
+        FakeVideoPlayerPlatform.kTestVideoDuration - const Duration(milliseconds: 500));
+      check(FakeVideoPlayerPlatform.position)
+        .equals(FakeVideoPlayerPlatform.kTestVideoDuration - const Duration(milliseconds: 500));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+
+      // At exactly the end of the video.
+      await tester.pump(const Duration(milliseconds: 500));
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.position).equals(FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.isPlaying).isFalse(); // stopped playing
+
+      // After the video ended.
+      await tester.pump(const Duration(milliseconds: 500));
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.position).equals(FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.isPlaying).isFalse();
+    });
+
+    testWidgets('ensure \'seekTo\' is called only once', (tester) async {
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestVideoUrl));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      const padding = 24.0;
+      final rect = tester.getRect(find.byType(Slider));
+      final trackWidth = rect.width - padding - padding;
+      final trackStartPos = rect.centerLeft + const Offset(padding, 0);
+      final twentyPercent = trackWidth * 0.2; // 20% increments
+
+      // Verify the actually displayed current position at each
+      // gesture increments.
+      final gesture = await tester.startGesture(trackStartPos);
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, Duration.zero);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(twentyPercent, 0.0));
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.2);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(twentyPercent, 0.0));
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.4);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(twentyPercent, 0.0));
```

Comment:
This test can be made a bit stronger by dragging the slider backward, as well as forward, before finally letting go. One could imagine bugs where either switching directions, or moving backward, triggers some different and wrong behavior.

(I'm getting this idea from just rereading my list at https://github.com/zulip/zulip-flutter/pull/587#discussion_r1597195082 .)